### ISSUE TITLE
tflint prodcuces errror for existing IAM instance profile

### ISSUE THREAD

**Issue by frezbo** [wata727/tflint#71]
I am a resource set like this:
```
resource "aws_instance" "ec2" {
	ami = "${var.ami}"
	instance_type = "${var.instance_type}"
	subnet_id = "${var.app_subnet}"
	vpc_security_group_ids = ["${aws_security_group.vnow_sg.id}"]
	key_name = "${var.ssh_key}"
	root_block_device {
		volume_type = "standard"
		volume_size = "${var.volume_size}"
		delete_on_termination = true
	}
	iam_instance_profile = "${var.instance_profile}"
	associate_public_ip_address = false
	user_data = "${data.template_file.user_data.rendered}"
	tags {
		Name = "${var.tag_name}"
		Owner = "${var.tag_owner}"
		ExpirationDate = "${var.tag_date}"
	}

}
```

In terraform.tfvars file I have set the instance profile. When i run tflint this is the output.
```
instance.tf
	ERROR:12 "s3accesstoec2" is invalid IAM profile name.

Result: 1 issues  (1 errors , 0 warnings , 0 notices)
```

but the instance profile already exists as per the aws cli command
```
aws iam list-instance-profiles | grep s3accesstoec2
```

O/P
```
                    "RoleName": "s3accesstoec2", 
                    "Arn": "arn:aws:iam::111111111111:role/s3accesstoec2"
            "InstanceProfileName": "s3accesstoec2", 
            "Arn": "arn:aws:iam::111111111111:instance-profile/s3accesstoec2"
```

**Comment by wata727** [comment]
Thanks your report. But I could not reproduce this problem... This detector call `ListInstanceProfiles` internally. After that, it will verify whether it is included in the obtained result.
Please run with `--debug` options and check whether to reproduce? When reproducing, paste the debug print.

**Comment by frezbo** [comment]
```
DEBUG:/loader.go:160 [INFO] Load HCL file: `.tflint.hcl`
DEBUG:/loader.go:100 [INFO] Load tfstate...
DEBUG:/loader.go:114 [INFO] Local state detected
DEBUG:/loader.go:130 [INFO] Load tfvars...
DEBUG:/loader.go:133 [INFO] Load `terraform.tfvars`
DEBUG:/loader.go:160 [INFO] Load HCL file: `terraform.tfvars`
DEBUG:/loader.go:160 [INFO] Load HCL file: `elb.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `elb_attach.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `instance.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `main.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `outputs.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `user_data.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `variables.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `vnow_elb_sg_rules.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `vnow_elg_sg.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `vnow_sg.tf`
DEBUG:/loader.go:160 [INFO] Load HCL file: `vnow_sg_rules.tf`
DEBUG:/aws.go:53 [INFO] set AWS region
DEBUG:/detector.go:143 [INFO] detect by `aws_db_instance_invalid_vpc_security_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_security_group_duplicate_name`
DEBUG:/detector.go:143 [INFO] detect by `aws_alb_duplicate_name`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_elasticache_cluster_invalid_type`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_elasticache_cluster_previous_type`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_db_instance_default_parameter_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_db_instance_previous_type`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_instance_invalid_type`
DEBUG:/detector.go:143 [INFO] detect by `aws_instance_previous_type`
DEBUG:/detector.go:143 [INFO] detect by `aws_elb_invalid_instance`
DEBUG:/aws_elb_invalid_instance.go:55 [ERROR] ERROR: key `instances` not found
DEBUG:/aws_elb_invalid_instance.go:58 [ERROR] ERROR: key `instances` not found
DEBUG:/detector.go:143 [INFO] detect by `aws_db_instance_invalid_db_subnet_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_instance_invalid_iam_profile`
DEBUG:/detector.go:143 [INFO] detect by `aws_elb_duplicate_name`
DEBUG:/detector.go:143 [INFO] detect by `aws_elasticache_cluster_default_parameter_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_elasticache_cluster_duplicate_id`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_alb_invalid_subnet`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_elb_invalid_subnet`
DEBUG:/aws_elb_invalid_subnet.go:53 [ERROR] ERROR: `subnets` value is not literal
DEBUG:/detector.go:143 [INFO] detect by `aws_db_instance_invalid_parameter_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_db_instance_duplicate_identifier`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_db_instance_readable_password`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_elasticache_cluster_invalid_parameter_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_elasticache_cluster_invalid_security_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_instance_not_specified_iam_profile`
DEBUG:/detector.go:143 [INFO] detect by `aws_instance_invalid_key_name`
DEBUG:/detector.go:143 [INFO] detect by `aws_alb_invalid_security_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_elb_invalid_security_group`
DEBUG:/aws_elb_invalid_security_group.go:53 [ERROR] ERROR: `security_groups` value is not literal
DEBUG:/aws_elb_invalid_security_group.go:64 [ERROR] ERROR: `"${aws_security_group.vnow_elb_sg.id}"` is not evaluable
DEBUG:/detector.go:143 [INFO] detect by `aws_db_instance_invalid_option_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_db_instance_invalid_type`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_elasticache_cluster_invalid_subnet_group`
DEBUG:/detector.go:247 [INFO] target resources are not found.
DEBUG:/detector.go:177 [INFO] skip this rule.
DEBUG:/detector.go:143 [INFO] detect by `aws_instance_default_standard_volume`
DEBUG:/detector.go:140 [INFO] ignore rule `aws_instance_invalid_ami`
DEBUG:/detector.go:143 [INFO] detect by `aws_instance_invalid_subnet`
DEBUG:/detector.go:143 [INFO] detect by `aws_instance_invalid_vpc_security_group`
DEBUG:/aws_instance_invalid_vpc_security_group.go:53 [ERROR] ERROR: `vpc_security_group_ids` value is not literal
DEBUG:/aws_instance_invalid_vpc_security_group.go:64 [ERROR] ERROR: `"${aws_security_group.vnow_sg.id}"` is not evaluable
instance.tf
	ERROR:12 "s3accesstoec2" is invalid IAM profile name.

Result: 1 issues  (1 errors , 0 warnings , 0 notices)

```
@wata727 attaching debug logs

**Comment by wata727** [comment]
Thanks. But I can not find a problem from the debug logs.
Probably, the cause of this problem is wrong AWS credentials. If you do not set credentials in `.tflint.hcl`, environment variables or default credentials are used. 
Please check that it is the same as the credentials used in the aws cli.

**Comment by frezbo** [comment]
@wata727  I have used the correct profile and all other tests are validating. I am able to successfully deploy the environment using terraform. I think the code is checking for IAM role instead of IAM instance profile.

**Comment by wata727** [comment]
TFLint calls `ListInstanceProfiles` instead of `ListRoles`. Therefore, it does not check for IAM role.
It is difficult to reproduce to me, so please tell me if you know something else.

**Comment by frezbo** [comment]
@wata727 I am not able to reproduce the issue in a new environment, but my old environment still shows the same error, even though its deployed successfully.
